File: android/src/main/java/com/stakkd/model/DemographicsAppendResponseDemographicsPersonWorkerType.java
/**
 * Stakkd API
 * # Introduction The Stakkd platform provides comprehensive data services that meet the business objectives of organizations ranging from Fortune 500 companies to startups. Our clients work in an array of industries, including insurance, financial, legal, travel, hospitality, retail, health, media, and telecommunications.  The Stakkd API is documented using the Open API 3.0.1 standard.  All endpoints are available at https://api.stakkd.io/v1. Detailed documentation for each endpoint is below, including the full URL for the endpoint.  # Account Types The API supports two different account types, each with different permissions.  ## Paid Account Paid accounts provide unlimited access to all of the URIs.  ## Trial Account Trial Accounts have a limited number of queries to the APIs. Keep in mind that queries which do not return results still count against the Trial Account query limits. Trial Accounts are temporary, but can be upgraded or extended by contacting [support@stakkd.io](mailto:support@stakkd.io) 
 *
 * The version of the OpenAPI document: v1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

package com.stakkd.model;

import io.swagger.annotations.*;
import com.google.gson.annotations.SerializedName;

@ApiModel(description = "")
public class DemographicsAppendResponseDemographicsPersonWorkerType {
  
  @SerializedName("percentWhiteCollar")
  private Integer percentWhiteCollar = null;
  @SerializedName("percentBlueCollar")
  private Integer percentBlueCollar = null;
  @SerializedName("percentOther")
  private Integer percentOther = null;

  /**
   * The Penetration Percentage for indicating whether the person is a white collar worker, +/- 3%.
   **/
  @ApiModelProperty(value = "The Penetration Percentage for indicating whether the person is a white collar worker, +/- 3%.")
  public Integer getPercentWhiteCollar() {
    return percentWhiteCollar;
  }
  public void setPercentWhiteCollar(Integer percentWhiteCollar) {
    this.percentWhiteCollar = percentWhiteCollar;
  }

  /**
   * The Penetration Percentage for indicating whether the person is a blue collar worker, +/- 3%.
   **/
  @ApiModelProperty(value = "The Penetration Percentage for indicating whether the person is a blue collar worker, +/- 3%.")
  public Integer getPercentBlueCollar() {
    return percentBlueCollar;
  }
  public void setPercentBlueCollar(Integer percentBlueCollar) {
    this.percentBlueCollar = percentBlueCollar;
  }

  /**
   * The Penetration Percentage for indicating whether the person is some other type of worker, +/- 3%.
   **/
  @ApiModelProperty(value = "The Penetration Percentage for indicating whether the person is some other type of worker, +/- 3%.")
  public Integer getPercentOther() {
    return percentOther;
  }
  public void setPercentOther(Integer percentOther) {
    this.percentOther = percentOther;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DemographicsAppendResponseDemographicsPersonWorkerType demographicsAppendResponseDemographicsPersonWorkerType = (DemographicsAppendResponseDemographicsPersonWorkerType) o;
    return (this.percentWhiteCollar == null ? demographicsAppendResponseDemographicsPersonWorkerType.percentWhiteCollar == null : this.percentWhiteCollar.equals(demographicsAppendResponseDemographicsPersonWorkerType.percentWhiteCollar)) &&
        (this.percentBlueCollar == null ? demographicsAppendResponseDemographicsPersonWorkerType.percentBlueCollar == null : this.percentBlueCollar.equals(demographicsAppendResponseDemographicsPersonWorkerType.percentBlueCollar)) &&
        (this.percentOther == null ? demographicsAppendResponseDemographicsPersonWorkerType.percentOther == null : this.percentOther.equals(demographicsAppendResponseDemographicsPersonWorkerType.percentOther));
  }

  @Override
  public int hashCode() {
    int result = 17;
    result = 31 * result + (this.percentWhiteCollar == null ? 0: this.percentWhiteCollar.hashCode());
    result = 31 * result + (this.percentBlueCollar == null ? 0: this.percentBlueCollar.hashCode());
    result = 31 * result + (this.percentOther == null ? 0: this.percentOther.hashCode());
    return result;
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class DemographicsAppendResponseDemographicsPersonWorkerType {\n");
    
    sb.append("  percentWhiteCollar: ").append(percentWhiteCollar).append("\n");
    sb.append("  percentBlueCollar: ").append(percentBlueCollar).append("\n");
    sb.append("  percentOther: ").append(percentOther).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}

File: php/lib/Stakkd/model/DemographicsAppendResponseDemographicsPersonWorkerType.php
<?php
/**
 * DemographicsAppendResponseDemographicsPersonWorkerType
 *
 * PHP version 7.3
 *
 * @category Class
 * @package  OpenAPI\Client
 * @author   OpenAPI Generator team
 * @link     https://openapi-generator.tech
 */

/**
 * Stakkd API
 *
 * # Introduction The Stakkd platform provides comprehensive data services that meet the business objectives of organizations ranging from Fortune 500 companies to startups. Our clients work in an array of industries, including insurance, financial, legal, travel, hospitality, retail, health, media, and telecommunications.  The Stakkd API is documented using the Open API 3.0.1 standard.  All endpoints are available at https://api.stakkd.io/v1. Detailed documentation for each endpoint is below, including the full URL for the endpoint.  # Account Types The API supports two different account types, each with different permissions.  ## Paid Account Paid accounts provide unlimited access to all of the URIs.  ## Trial Account Trial Accounts have a limited number of queries to the APIs. Keep in mind that queries which do not return results still count against the Trial Account query limits. Trial Accounts are temporary, but can be upgraded or extended by contacting [support@stakkd.io](mailto:support@stakkd.io)
 *
 * The version of the OpenAPI document: v1
 * Generated by: https://openapi-generator.tech
 * OpenAPI Generator version: 5.4.0
 */

/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

namespace OpenAPI\Client\Stakkd.model;

use \ArrayAccess;
use \OpenAPI\Client\ObjectSerializer;

/**
 * DemographicsAppendResponseDemographicsPersonWorkerType Class Doc Comment
 *
 * @category Class
 * @package  OpenAPI\Client
 * @author   OpenAPI Generator team
 * @link     https://openapi-generator.tech
 * @implements \ArrayAccess<TKey, TValue>
 * @template TKey int|null
 * @template TValue mixed|null
 */
class DemographicsAppendResponseDemographicsPersonWorkerType implements ModelInterface, ArrayAccess, \JsonSerializable
{
    public const DISCRIMINATOR = null;

    /**
      * The original name of the model.
      *
      * @var string
      */
    protected static $openAPIModelName = 'demographicsAppendResponse_demographics_person_workerType';

    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $openAPITypes = [
        'percent_white_collar' => 'int',
        'percent_blue_collar' => 'int',
        'percent_other' => 'int'
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      * @phpstan-var array<string, string|null>
      * @psalm-var array<string, string|null>
      */
    protected static $openAPIFormats = [
        'percent_white_collar' => null,
        'percent_blue_collar' => null,
        'percent_other' => null
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function openAPITypes()
    {
        return self::$openAPITypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function openAPIFormats()
    {
        return self::$openAPIFormats;
    }

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @var string[]
     */
    protected static $attributeMap = [
        'percent_white_collar' => 'percentWhiteCollar',
        'percent_blue_collar' => 'percentBlueCollar',
        'percent_other' => 'percentOther'
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'percent_white_collar' => 'setPercentWhiteCollar',
        'percent_blue_collar' => 'setPercentBlueCollar',
        'percent_other' => 'setPercentOther'
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'percent_white_collar' => 'getPercentWhiteCollar',
        'percent_blue_collar' => 'getPercentBlueCollar',
        'percent_other' => 'getPercentOther'
    ];

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @return array
     */
    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * The original name of the model.
     *
     * @return string
     */
    public function getModelName()
    {
        return self::$openAPIModelName;
    }


    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     *                      initializing the model
     */
    public function __construct(array $data = null)
    {
        $this->container['percent_white_collar'] = $data['percent_white_collar'] ?? null;
        $this->container['percent_blue_collar'] = $data['percent_blue_collar'] ?? null;
        $this->container['percent_other'] = $data['percent_other'] ?? null;
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }


    /**
     * Gets percent_white_collar
     *
     * @return int|null
     */
    public function getPercentWhiteCollar()
    {
        return $this->container['percent_white_collar'];
    }

    /**
     * Sets percent_white_collar
     *
     * @param int|null $percent_white_collar The Penetration Percentage for indicating whether the person is a white collar worker, +/- 3%.
     *
     * @return self
     */
    public function setPercentWhiteCollar($percent_white_collar)
    {
        $this->container['percent_white_collar'] = $percent_white_collar;

        return $this;
    }

    /**
     * Gets percent_blue_collar
     *
     * @return int|null
     */
    public function getPercentBlueCollar()
    {
        return $this->container['percent_blue_collar'];
    }

    /**
     * Sets percent_blue_collar
     *
     * @param int|null $percent_blue_collar The Penetration Percentage for indicating whether the person is a blue collar worker, +/- 3%.
     *
     * @return self
     */
    public function setPercentBlueCollar($percent_blue_collar)
    {
        $this->container['percent_blue_collar'] = $percent_blue_collar;

        return $this;
    }

    /**
     * Gets percent_other
     *
     * @return int|null
     */
    public function getPercentOther()
    {
        return $this->container['percent_other'];
    }

    /**
     * Sets percent_other
     *
     * @param int|null $percent_other The Penetration Percentage for indicating whether the person is some other type of worker, +/- 3%.
     *
     * @return self
     */
    public function setPercentOther($percent_other)
    {
        $this->container['percent_other'] = $percent_other;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param integer $offset Offset
     *
     * @return boolean
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param integer $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param integer $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    /**
     * Serializes the object to a value that can be serialized natively by json_encode().
     * @link https://www.php.net/manual/en/jsonserializable.jsonserialize.php
     *
     * @return mixed Returns data which can be serialized by json_encode(), which is a value
     * of any type other than a resource.
     */
    public function jsonSerialize()
    {
       return ObjectSerializer::sanitizeForSerialization($this);
    }

    /**
     * Gets the string presentation of the object
     *
     * @return string
     */
    public function __toString()
    {
        return json_encode(
            ObjectSerializer::sanitizeForSerialization($this),
            JSON_PRETTY_PRINT
        );
    }

    /**
     * Gets a header-safe presentation of the object
     *
     * @return string
     */
    public function toHeaderValue()
    {
        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}


